# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Ibrahim-1rfan/Flyrank-/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*


I have decided upon Lane 2: Refresh / Content Opportunity Scoring. Content teams often will spend dozens of hours manually auditing sites or blindly rewriting articles without knowing where changes will actually deliver results. I chose this lane to build a priority queue that ranks decaying, high-traffic pages so editors can immediately focus their time on updates with the highest traffic recovery potential.

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*


What I hope is improving on deciding which top 20 to 50 mature articles the editorial team should audit and rewrite first each month.

This will help Content Operations Managers, SEO Specialists, and Copywriters.

Making a wrong recommendation either wastes hours of copywriter budget on healthy articles that do not need work, or misses declining high-value pages and allows them to quietly lose search rankings to competitors. I am framing this as a Ranking Task measured by Precision@K, comparing how effectively my model prioritizes real opportunities against a simple rule-based baseline.

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

# Load dataset
df = pd.read_csv("/content/content_refresh_anonymized.csv")

# Key metrics
total_pages = len(df)
declining_pages = len(df[df["trend_direction"] == "down"])
declining_pct = (declining_pages / total_pages) * 100
high_impact_pages = len(
    df[(df["trend_direction"] == "down") & (df["impressions_90d"] >= 1000)]
)

print(f"Total articles: {total_pages:,}")
print(f"Declining articles: {declining_pages:,} ({declining_pct:.1f}%)")
print(f"High-impact targets (>=1,000 impressions): {high_impact_pages:,}")

Total articles: 30,000
Declining articles: 16,262 (54.2%)
High-impact targets (>=1,000 impressions): 8,031


Analyzing the 30,000 mature articles in the starter dataset shows that 16,262 pages (54.2%) are currently in an active downward traffic trend. Out of those declining pages, 8,031 are high-impact assets that generated at least 1,000 impressions over the last 90 days. Because an editorial team cannot manually inspect over 8,000 decaying pages, an automated priority ranking system is essential.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

I can provide a decision-support priority score that helps human teams allocate their review time efficiently, measure observed historical traffic drops using safe dataset features, and demonstrate whether my machine learning model outperforms a standard rule baseline using Precision@K. However, I cannot claim that updating an article guarantees traffic recovery, predict Google's internal algorithm adjustments, or replace human editorial judgment regarding how the content should be rewritten.

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.